index.js

```diff
@@ -0,0 +1,221 @@
+    }
+
+    /**
+     * Starts a s3m build
+     * @method _start
+     * @param  {Object}   config            A configuration object
+     * @param  {String}   config.buildId    ID for the build
+     * @param  {String}   config.container  Container for the build to run in
+     * @param  {String}   config.apiUri     API Uri
+     * @param  {String}   config.token      JWT for the Build
+     * @return {Promise}
+     */
+    _start(config) {
+        return this._createContainer(
+            {
+                name: `${config.buildId}-init`,
+                Image: `screwdrivercd/launcher:${this.launchVersion}`,
+                Entrypoint: '/bin/true',
+                Labels: {
+                    sdbuild: config.buildId
+                }
+            })
+            .then(launchContainer => Promise.all([
+                this._createContainer({
+                    name: `${config.buildId}-log`,
+                    Image: `screwdrivercd/log-service:${this.logVersion}`,
+                    Entrypoint: '/opt/screwdriver/tini',
+                    Labels: {
+                        sdbuild: config.buildId
+                    },
+                    Cmd: [
+                        '--',
+                        '/opt/screwdriver/logservice',
+                        '--emitter',
+                        '/opt/screwdriver/emitter',
+                        config.buildId
+                    ],
+                    Env: [
+                        `SD_TOKEN=${config.token}`
+                    ],
+                    HostConfig: {
+                        Memory: 200 * 1024 * 1024,
+                        MemoryLimit: 300 * 1024 * 1024,
+                        VolumesFrom: [
+                            `${launchContainer.id}:rw`
+                        ]
+                    }
+                }),
+                this._createContainer({
+                    name: `${config.buildId}-build`,
+                    Image: config.container,
+                    Entrypoint: '/opt/screwdriver/tini',
+                    Labels: {
+                        sdbuild: config.buildId
+                    },
+                    Cmd: [
+                        '--',
+                        '/opt/screwdriver/launch',
+                        '--api-uri',
+                        config.apiUri,
+                        '--emitter',
+                        '/opt/screwdriver/emitter',
+                        config.buildId
+                    ],
+                    Env: [
+                        `SD_TOKEN=${config.token}`
+                    ],
+                    HostConfig: {
+                        Memory: 2 * 1024 * 1024 * 1024,
+                        MemoryLimit: 3 * 1024 * 1024 * 1024,
+                        VolumesFrom: [
+                            `${launchContainer.id}:rw`
+                        ]
+                    }
+                })
+            ]))
+            .then(([logContainer, buildContainer]) => Promise.all([
```

Comment:
Ahh this is the thing you were talking about. It would have to wait to create and start both containers.